modules/mbedtls/CMakeLists.txt

```diff
@@ -53,6 +52,10 @@ else()
   # Lib mbedtls_external depends on libgcc (I assume?) so to allow
   # mbedtls_external to link with gcc we need to ensure it is placed
   # after mbedtls_external on the linkers command line.
+else()
+  # If none of either CONFIG_MBEDTLS_BUILTIN or CONFIG_MBEDTLS_LIBRARY
+  # are defined the users need to manually add the mbedtls library and
+  # included the required directories for mbedtls in their projects.
```

Comment:
Let's help users realize how they can not have either defined. Something like:

> Also, a Kconfig fragment must add another choice to MBEDTLS_IMPLEMENTATION.

Reply:
That's a good point :+1:

Reply:
and mayby the title is a bit misleading:
```
modules: mbedtls: Provide 3rd option for MBEDTLS configuration
```
cause it doesn't provide a 3rd options, it only allows downstream users to provide out-of-tree choices and implementations.